python-frontend/src/main/java/org/sonar/python/tree/AnnotatedAssignmentImpl.java

```diff
@@ -31,20 +31,19 @@
 import org.sonar.plugins.python.api.tree.Token;
 import org.sonar.plugins.python.api.tree.TreeVisitor;
 import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.plugins.python.api.tree.TypeAnnotation;
 
 public class AnnotatedAssignmentImpl extends SimpleStatement implements AnnotatedAssignment {
   private final Expression variable;
-  private final Token colonToken;
-  private final Expression annotation;
   private final Token equalToken;
   private final Expression assignedValue;
   private final Separators separators;
+  private final TypeAnnotation annotation;
 
-  public AnnotatedAssignmentImpl(Expression variable, Token colonToken, Expression annotation,
+  public AnnotatedAssignmentImpl(Expression variable, Token colonToken, Expression typeAnnotation,
```

Comment:
I would rather put the instantiation part in PythonTreeMaker rather than passing two params.